presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkAdaptiveQueryExecution.java

```diff
@@ -247,6 +251,17 @@ protected List<Tuple2<MutablePartitionId, PrestoSparkSerializedPage>> doExecute(
         log.info("Using AdaptiveQueryExecutor");
         log.info(format("Logical plan : %s",
                 textLogicalPlan(this.planAndMore.getPlan().getRoot(), this.planAndMore.getPlan().getTypes(), this.planAndMore.getPlan().getStatsAndCosts(), metadata.getFunctionAndTypeManager(), session, 0)));
+        queryMonitor.queryUpdatedEvent(
+                createQueryInfo(
+                        session,
+                        query,
+                        PLANNING,
+                        Optional.of(planAndMore),
+                        sparkQueueName,
+                        Optional.empty(),
+                        queryStateTimer,
+                        Optional.of(createStageInfo(session.getQueryId(), planFragmenter.fragmentQueryPlan(session, planAndMore.getPlan(), warningCollector), ImmutableList.of())),
```

Comment:
Looks like we are using the static plan fragmenter to display the fragmented plan. While this fixes the regression, I wonder if there are cases where we might display the wrong information. Might be an edge case but something to think about. 

I also don't see a easy to fix this edge case, because even if we somehow figured out a way to publish few fragments at a time, the QueryInfo object expects a top down fragmented plan v/s bottom up plan.

Reply:
Thats true. It may show stale info when runtime optimization kicks off which will get updated once we have completion event. Currently it just captures the plan we start with and the final plan once query completes.